### PR TITLE
convert provider stanza to use new style config

### PR DESCRIPTION
**Pull request by coffeepac**
WIP:  need another commit to convert the GKE provider

~~the dry_run role will set a global fact named dry_run to be true.  we are also now defaulting
this global fact to be false.  this allows us to only run part of a role when the dry_run tag
is set.  currently, tags can only be applied at the role level.~~
removed the dry_run role as it didn't work.  I had a typo that was kind of making it work but when I fixed the typo things fell apart.

AWS provider conversion:  this includes converting aws tasks and templates.  this also
includes converting the selector role for providers from a list of dependency roles to using
the new (as of ansible 2.2) preview module (beta in most parlance) named 'include_role' which
allows the developer to use roles like tasks, including the ability to loop over role executions.
which is required for a list of clusters where each cluster can have a different provider.  the
GKE selector has been ported but I don't think anything good can happen if you try to use GKE
with this PR as-is.  I think there is some mutually exclusive config going on.

This handles #269

### CODE REVIEW

File: ansible/roles/kraken.ssh/kraken.ssh.aws/tasks/main.yaml [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
+- set_fact:
+    cluster: "{{ cluster_node_tuple }}"
+
 - name: Gather inventory of all cluster nodes
   ec2_remote_facts:
-    region: "{{ kraken_config.providerConfig.region }}"
-    aws_access_key: "{{ kraken_config.providerConfig.authentication.accessKey or omit }}"
-    aws_secret_key: "{{ kraken_config.providerConfig.authentication.accessSecret or omit }}"
-    profile: "{{ kraken_config.providerConfig.authentication.credentialsProfile or omit }}"
+    region: "{{ config.providerConfig.region }}"
```
Comment (davidewatson) [inline ansible/roles/kraken.ssh/kraken.ssh.aws/tasks/main.yaml]: s/config/cluster/g?

File: ansible/roles/kraken.provider/kraken.provider.aws/tasks/main.yaml [hunk 2]
```diff
@@ -1,4 +1,7 @@
 ---
+- set_fact:
+    cluster: "{{ cluster_node_tuple }}"
```
Comment (davidewatson) [inline ansible/roles/kraken.provider/kraken.provider.aws/tasks/main.yaml]: Or leave it the same as elsewhere for consistency, and I can rename it later?

File: ansible/roles/kraken.ssh/kraken.ssh.aws/tasks/main.yaml (hunk 1 above)
Comment (coffeepac) [inline ansible/roles/kraken.ssh/kraken.ssh.aws/tasks/main.yaml]: yeah, there are a bunch of these.  My typing skills were a bit muddled this afternoon.  I'll have another PR up in a bit, struggling with keyPair creation now.

File: ansible/roles/kraken.provider/kraken.provider.aws/tasks/main.yaml (hunk 2 above)
Comment (coffeepac) [inline ansible/roles/kraken.provider/kraken.provider.aws/tasks/main.yaml]: haha, yeah, its neither a tuple nor does it include cluster.  I'm happy to set it to cluster in the top level.  its consistent but dumb on my part.

**Comment by coffeepac** [comment]
@davidewatson there's something wonky with using cluster as loop_variable, I'm not sure what but when I did use cluster as the loop var in the selector roles I got a warning about what seemed to be a scoping issue.  Because include_role is preview I am erring on the side of caution and putting in an extra level of caution.

I can gen the warning tomorrow if you'd like.  Only takes 5 minutes.

Otherwise, this code works now for both aws and gke providers.  I have sample configs for both aws and gke as well I can share for anyone who wants 'em.

**Pull request merged**